Spigot-API-Patches/0243-Added-SoundBuilder.patch

```diff
@@ -0,0 +1,331 @@
++     * world
++     * @return a reference to this object.
++     */
++    @NotNull
++    SoundBuilder receivers(@Nullable Player... receivers);
++
++    /**
++     * Selects all players within a cuboid selection around the sound location, within the
++     * specified bounding box. If you want a more spherical check, see {@link #receivers(int,
++     * boolean)}
++     *
++     * @param radius amount to add on all axis
++     * @return a reference to this object.
++     */
++    @NotNull
++    SoundBuilder receivers(int radius);
++
++    /**
++     * Selects all players within the specified radius around the sound location. If byDistance is
++     * false, behavior uses cuboid selection the same as {@link #receivers(int, int)} If byDistance is
++     * true, radius is tested by distance in a spherical shape
++     *
++     * @param radius amount to add on each axis
++     * @param byDistance true to use a spherical radius, false to use a cuboid
++     * @return a reference to this object.
++     */
++    @NotNull
++    SoundBuilder receivers(int radius, boolean byDistance);
++
++    /**
++     * Selects all players within a cuboid selection around the sound location, within the
++     * specified bounding box. Allows specifying a different Y size than X and Z If you want a more
++     * cylinder check, see {@link #receivers(int, int, boolean)} If you want a more spherical check,
++     * see {@link #receivers(int, boolean)}
++     *
++     * @param xzRadius amount to add on the x/z axis
++     * @param yRadius amount to add on the y axis
++     * @return a reference to this object.
++     */
++    @NotNull
++    SoundBuilder receivers(int xzRadius, int yRadius);
++
++    /**
++     * Selects all players within the specified radius around the sound. If byDistance is
++     * false, behavior uses cuboid selection the same as {@link #receivers(int, int)} If byDistance is
++     * true, radius is tested by distance on the y plane and on the x/z plane, in a cylinder shape.
++     *
++     * @param xzRadius amount to add on the x/z axis
++     * @param yRadius amount to add on the y axis
++     * @param byDistance true to use a cylinder shape, false to use cuboid
++     * @return a reference to this object.
++     */
++    @NotNull
++    SoundBuilder receivers(int xzRadius, int yRadius, boolean byDistance);
++
++    /**
++     * Selects all players within a cuboid selection around the sound location, within the
++     * specified bounding box. If you want a more cylinder check, see {@link #receivers(int, int,
++     * boolean)} If you want a more spherical check, see {@link #receivers(int, boolean)}
++     *
++     * @param xRadius amount to add on the x axis
++     * @param yRadius amount to add on the y axis
++     * @param zRadius amount to add on the z axis
++     * @return a reference to this object.
++     */
++    @NotNull
++    SoundBuilder receivers(int xRadius, int yRadius, int zRadius);
```

Comment:
I get it that you are trying to copy the `ParticleBuilder` API, but having this many methods with the same name all taking similar, but slightly different parameters is confusing. I would suggest using better method names.

Reply:
They all setting up the receivers, I think `receivers` is an great name for that!

Reply:
I disagree. One is spherical, another is cuboidal: very different expectations with unclear parameter names to differ them.